File: gin-admin-backend/internal/data/menu.go
package data

import (
	"time"
)

type Menu struct {
	ID         string    `gorm:"primaryKey;type:varchar(36)" json:"id"`
	ParentID   string    `gorm:"type:varchar(255)" json:"parentId"`
	Path       string    `gorm:"type:varchar(255)" json:"path"`
	Component  string    `gorm:"type:varchar(255)" json:"component"`
	Redirect   string    `gorm:"type:varchar(255)" json:"redirect"`
	Title      string    `gorm:"type:varchar(100);not null" json:"title"`
	Name       string    `gorm:"type:varchar(100)" json:"name"`
	Icon       string    `gorm:"type:varchar(50)" json:"icon"`
	Meta       string    `gorm:"type:json" json:"meta"` // Stored as JSON string
	Type       int       `gorm:"type:tinyint;default:0;comment:0:Directory,1:Menu,2:Button" json:"type"`
	Status     int       `gorm:"type:tinyint;default:1" json:"status"`
	Permission string    `gorm:"type:varchar(100)" json:"permission"`
	Sort       int       `gorm:"type:int;default:0" json:"sort"`
	CreatedAt  time.Time `json:"createdAt"`
	UpdatedAt  time.Time `json:"updatedAt"`
}

func (Menu) TableName() string {
	return "sys_menu"
}

File: python-admin-backend/app/models/menu.py
from sqlalchemy import Column, String, Integer, DateTime, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from sqlalchemy.dialects.mysql import JSON
from app.db.database import Base
from app.models import sys_role_menu


class Menu(Base):
    __tablename__ = "sys_menu"

    id = Column(String(36), primary_key=True, index=True)
    parent_id = Column(String(36), nullable=True)
    path = Column(String(255), nullable=True)
    component = Column(String(255), nullable=True)
    redirect = Column(String(255), nullable=True)
    title = Column(String(100), nullable=False)
    name = Column(String(100), nullable=True)
    icon = Column(String(50), nullable=True)
    meta = Column(JSON, nullable=True)
    type = Column(Integer, server_default="0")
    status = Column(Integer, server_default="1")
    permission = Column(String(100), nullable=True)
    sort = Column(Integer, server_default="0")
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
    roles = relationship("Role", secondary=sys_role_menu, back_populates="menus")
